docs/user/openstack/observability.md

```diff
@@ -0,0 +1,381 @@
+them with mTLS (mutual TLS). The target Prometheus needs to be configured to
+[accept client TLS certificates][prometheus-mtls], and
+[Remote-Write][prometheus-remote-write-receiver-flag].
+
+[prometheus-mtls]: https://prometheus.io/docs/prometheus/latest/configuration/https/
+[prometheus-remote-write-receiver-flag]: https://prometheus.io/docs/prometheus/latest/feature_flags/#remote-write-receiver "Prometheus feature flags: Remote-Write receiver"
+
+
+<!--
+To generate test certificates:
+
+```bash
+# Generate a CA if you don't have one already
+openssl genrsa -out ca.key 4096
+openssl req -batch -new -x509 -key ca.key -out ca.crt
+
+# Generate the client certificates and sign them:
+for target in server ocp-client osp-client; do
+    openssl genrsa -out "${target}.key" 4096
+    openssl req -batch -new -key "${target}.key" -out "${target}.csr"
+    openssl x509 -req -CA ca.crt -CAkey ca.key -CAcreateserial -in "${target}.csr" -out "${target}.crt"
+done
+```
+-->
+
+<!--
+For testing purpose, we can do the following to set up basic auth in addition
+or in stead of mTLS:
+
+1. Provision a Fedora VM
+2. Install `dnf install golang-github-prometheus caddy`
+3. Configure prometheus to enable remote write (and limit retention to avoid
+   filling up disk space). In `/etc/default/prometheus`, add the following
+   line:
+
+```
+ARGS='--enable-feature=remote-write-receiver --storage.tsdb.retention.time=1d'
+```
+
+1. Enable and restart the Prometheus systemd unit
+2. Add a security group rule to allow HTTPS (port 443)
+3. Setup Caddy with (`/etc/caddy/Caddyfile`):
+
+```Caddyfile
+https://external-prometheus.example {
+
+    basicauth {
+        # caddy hash-password
+        user hashed-password
+    }
+
+    reverse_proxy http://localhost:9090
+}
+```
+-->
+
+
+
+We will assume that the external Prometheus is reachable at the URL
+`https://external-prometheus.example`.
+
+#### Set up remote-write from RHOSO's telemetry-operator
+
+Telemetry should be enabled in the RHOSO environment. If it is not the case,
+refer to the
+[documentation](https://docs.redhat.com/en/documentation/red_hat_openstack_services_on_openshift/18.0/html/customizing_the_red_hat_openstack_services_on_openshift_deployment/rhoso-observability_custom_dataplane#rhoso-observability_rhoso-observability).
+
+<!--
+Essentially, enabling telemetry boils down to flipping a property of the
+openstackcontrolplane object:
+
+```bash
+oc patch OpenStackControlPlane/controlplane --type merge -p '{"spec":{"telemetry":{"enabled": true, "template":{"ceilometer":{"enabled": true}}}}}'
+```
+-->
+
+> [!NOTE]
+Make sure you have the Cluster Observability Operator installed in the
+OpenShift cluster running the OpenStack control plane, as this is a requirement
+for the OpenStack Telemetry Operator. Follow [these
+directions](https://github.com/openstack-k8s-operators/architecture/blob/main/examples/dt/uni01alpha/control-plane.md#cluster-observability-operator)
```

Comment:
Ideally we would point to the osasinfra DT, but https://github.com/openstack-k8s-operators/architecture/pull/418 hasn't merged yet.

Reply:
Can you please point me to the osasinfra DT?